ui/src/views/infra/zone/ZoneWizardRegisterTemplate.vue

```diff
@@ -0,0 +1,339 @@
+          </template>
+        </template>
+      </a-table>
+      <div class="form-action">
+        <a-button class="button-back" @click="handleDone">{{ selectedRowKeys.length > 0 ? $t('label.done') : $t('label.skip') }}</a-button>
+        <a-button class="button-next" type="primary" @click="handleSubmit" ref="submit">{{ $t('label.register.template') }}</a-button>
+      </div>
+    </div>
+  </template>
+
+<script>
+import { api } from '@/api'
+import { genericCompare } from '@/utils/sort.js'
+import OsLogo from '@/components/widgets/OsLogo'
+
+export default {
+  name: 'ZoneWizardRegisterTemplate',
+  components: {
+    OsLogo
+  },
+  props: {
+    zoneid: {
+      type: String,
+      required: false
+    }
+  },
+  data: () => ({
+    columns: null,
+    loading: false,
+    predefinedTemplates: null,
+    rowKey: 0,
+    selectedRowKeys: [],
+    defaultOsTypeId: null,
+    deployedTemplates: {}
+  }),
+  created () {
+    this.initForm()
+  },
+  computed: {
+    rowSelection () {
+      return {
+        selectedRowKeys: this.selectedRowKeys || [],
+        onChange: this.onSelectRow,
+        getCheckboxProps: (record) => {
+          return {
+            disabled: record.disabled
+          }
+        }
+      }
+    }
+  },
+  methods: {
+    async initForm () {
+      this.columns = [
+        {
+          title: 'Name',
+          dataIndex: 'name',
+          key: 'name',
+          width: 240,
+          sorter: (a, b) => { return genericCompare(a.name || '', b.name || '') }
+        },
+        {
+          title: 'Arch',
+          dataIndex: 'arch',
+          key: 'arch',
+          width: 80,
+          sorter: (a, b) => { return genericCompare(a.arch || '', b.arch || '') }
+        },
+        {
+          title: 'URL',
+          dataIndex: 'url',
+          key: 'url'
+        }
+      ]
+      this.predefinedTemplates = [
+        {
+          id: 1,
+          name: 'Ubuntu 24.04',
+          version: '24.04',
+          osCategoryName: 'Ubuntu',
+          arch: 'x86_64',
+          url: 'https://cloud-images.ubuntu.com/releases/noble/release/ubuntu-24.04-server-cloudimg-amd64.img'
+        },
+        {
+          id: 2,
+          name: 'Ubuntu 22.04',
+          version: '22.04',
+          osCategoryName: 'Ubuntu',
+          arch: 'x86_64',
+          url: 'https://cloud-images.ubuntu.com/releases/jammy/release/ubuntu-22.04-server-cloudimg-amd64.img'
+        },
+        {
+          id: 3,
+          name: 'Ubuntu 20.04',
+          version: '20.04',
+          osCategoryName: 'Ubuntu',
+          arch: 'x86_64',
+          url: 'https://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64.img'
+        },
+        {
+          id: 4,
+          name: 'Debian GNU/Linux 12 (64-bit)',
+          version: '12',
+          osCategoryName: 'Debian',
+          arch: 'x86_64',
+          url: 'https://laotzu.ftp.acc.umu.se/images/cloud/bookworm/latest/debian-12-genericcloud-amd64.qcow2'
```

Comment:
better to download from official site instead of a mirror
https://cloud.debian.org/images/cloud/bookworm/latest/